core/src/main/java/org/mule/runtime/core/internal/streaming/CursorManager.java

```diff
@@ -141,35 +128,31 @@ private void registerEventContext(BaseEventContext eventContext) {
     private AtomicBoolean disposed = new AtomicBoolean(false);
     private AtomicInteger cursorCount = new AtomicInteger(0);
 
-    private final LoadingCache<CursorProvider, Set<Cursor>> cursors = CacheBuilder.newBuilder()
-        .removalListener((RemovalListener<CursorProvider, Set<Cursor>>) notification -> {
+    private final LoadingCache<CursorProvider, Set<Cursor>> cursors = Caffeine.newBuilder()
+        .removalListener((key, value, cause) -> {
```

Comment:
FYI, you can add types to a lambda, e.g. 
`(CursorProvider key, Collection<Cursor> value, RemovalCause cause) -> ...`
which then propagates the types into later build stages.